packages/core/src/textFields/interaction/commands/RemoveWordCommand.ts

```diff
@@ -0,0 +1,78 @@
+     * @param targetField The text field to remove the text from
+     * @param direction The direction to remove the word (-1 == backwards)
+     */
+    public constructor(
+        textField: ITextField,
+        direction: IRetrievableArgument<number> = -1
+    ) {
+        super(textField, ({selection}) => {
+            direction = retrieveArgument(direction);
+            const start = Math.min(selection.start, selection.end);
+            const end = Math.max(selection.start, selection.end);
+
+            let newCaretPos = start;
+            let alteration: ITextAlterationInput;
+
+            // If text is currently selected, ignore the direction and just remove it
+            if (start != end) alteration = {start, end, content: ""};
+            // If the direction is backwards remove before the cursor till whitespace and move the cursor backwards
+            else if (direction < 0) {
+                let startIndex = this.backwardsSearchString(textField.get(), start);
+                alteration = {start: startIndex, end, content: ""};
+                newCaretPos = startIndex;
+            }
+            // If the direction is forwards remove after the cursor till whitespace
+            else {
+                let endIndex = this.forwardsSearchString(textField.get(), start);
+                endIndex = endIndex >= 0 ? endIndex : textField.get().length;
+                alteration = {start: start, end: endIndex, content: ""};
+            }
+
+            // Return the new changes and new selection
+            return {text: alteration, selection: {start: newCaretPos, end: newCaretPos}};
+        });
+    }
+
+    /**
+     * Backwards searches the string for the position next word including all spaces around it
+     * @param text The text to search in
+     * @param startIndex The start index from where the search will start
+     * @returns The end index
+     */
+    private backwardsSearchString(text: string, startIndex: number): number {
+        text = text.split("").reverse().join("");
+        var reversedStartIndex = text.length - startIndex;
+        var endIndex = this.forwardsSearchString(text, reversedStartIndex);
+        if (endIndex < 0) return -1;
+
+        return text.length - endIndex;
+    }
+
+    /**
+     * Forward searches the string for the position next word including all spaces around it
+     * @param text The text to search in
+     * @param startIndex The start index from where the search will start
+     * @returns The end index
+     */
+    private forwardsSearchString(text: string, startIndex: number): number {
+        var searchStartIndex = startIndex + text.substr(startIndex).search(/[^\s]/);
+        var firstSpaceIndex = text.substr(searchStartIndex).search(/\s/);
+        if (firstSpaceIndex < 0) return -1;
+
+        var firstCharAfterSpace =
+            text.substr(searchStartIndex + firstSpaceIndex).search(/[^\s]/) - 1;
```

Comment:
@bramhoven  @TarVK  just had a thought about searching for `\s` or the lack of it... Can you check windows behaviour with strings like:

```
hello,world - stops at char
hello;word - stops at char
hello'world - continues through char
hello"world  - stops at char
hello-world - stops at char
hello_world - continues through char
hello[world - stops at char
hello]world - stops at char
hello{world - stops at char
hello}world - stops at char
hello(world - stops at char
hello)world - stops at char
hello\world - stops at char
hello|world - stops at char
hello/world - stops at char
hello?world - stops at char
hello!world - stops at char
hello`world - stops at char
hello~world - stops at char
hello<world - stops at char
hello>world - stops at char
hello.world - stops at char
hello,world - stops at char
hello=world - stops at char
hello+world - stops at char
hello@world - stops at char
hello£world - stops at char
helllo$world - stops at char
hello%world - stops at char
hello^world - stops at char
hello&world - stops at char
hello*world - stops at char
hello±world - stops at char
hello§world - stops at char
hello0world - continues through char
```

So on Mac at least `alt+left/right` and `alt+delete/backspace` both continue though the characters [A-Za-z0-9_'] stopping at all characters [^A-Za-z0-9_'].
1. How is it on windows?  
2. Would be useful to export this per platform.

Reply:
@sancarn I have now made it stop at only whitespaces. If we want to make it stop at those specific characters we have to change the searches.

Reply:
It seems like I actually had a function for this already, we should probably try to reuse it:
https://github.com/LaunchMenu/LaunchMenu/blob/9ed5afc7c393e3905651dbf18242c303ecc8a88d/packages/core/src/textFields/interaction/getJumpTokenPos.ts#L8

We can make any word jump related changes in there, and that will keep things consistent between the different jump types. (using "jump types" for the lack of a better)

Reply:
I did some analysis of different OS's software and where they stop at the char and where they do not, for those interested. 

![WordJumpingAnalysis](https://user-images.githubusercontent.com/7938900/124357606-5648cb80-dc14-11eb-8a9a-500d7d4ddd0c.png)

If you could reuse `getJumpTokenPos()` then we can change that at a later date if need be :)

Reply:
Alright, I will probably be able to implement the use of getJumpTokenPos tonight.